### ISSUE TITLE
Latest version of mockk (1.12.3) incompatible with kotlin 1.4.32

### ISSUE THREAD

**Issue by jbotuck** [mockk/mockk#794]
### Expected Behavior

My tests should compile

### Current Behavior

What is the current behavior?

They don't compile

### Failure Information (for bugs)

```
[ERROR] Incompatible classes were found in dependencies. Remove them from the classpath or use '-Xskip-metadata-version-check' to suppress errors
[ERROR] C:/Users/EI5246/.m2/repository/io/mockk/mockk-agent-api/1.12.3/mockk-agent-api-1.12.3.jar!/META-INF/mockk-agent-api.kotlin_module: (-1, -1) Module was compiled with an incompatible version of Kotlin. The binary version of it
s metadata is 1.6.0, expected version is 1.4.2.
```

**Comment by breun** [comment]
I also ran into this issue when trying to update from MockK 1.12.2 to 1.12.3 on a project which uses Kotlin 1.4.

**Comment by kingargyle** [comment]
Similar, 

Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.6.0, expected version is 1.4.2

If I go back to 1.12.0 - 1.12.2 everything works

**Comment by TWiStErRob** [comment]
Same here, just was about to report this.

REAMDE.md says:
> From version 1.10.0 MockK does not support Kotlin 1.2.*

Release changelog has 0 mention that you started compiling with Kotlin 1.6, which is a major change for a Kotlin library because of inlining and metadata.

I think you might be missing `languageVersion = "1.4"` ([docs](https://kotlinlang.org/docs/gradle.html#attributes-common-to-jvm-and-js)) at

https://github.com/mockk/mockk/blob/ddb5a99bbe4433392c339d713683339ecb064006/build.gradle#L37-L41

changed by https://github.com/mockk/mockk/commit/6f125ea8b4a14f7bbd207c419e13596692011a3b

**Comment by TWiStErRob** [comment]
Note: after fixing I recommend a tool to prevent this.
OkHttp does a really good job of being compatible. Here's their integrations:
japicmp (they moved away from it): https://github.com/square/okhttp/pull/7118
[Kotlin official plugin](https://github.com/Kotlin/binary-compatibility-validator) (in favor of this): https://github.com/square/okhttp/pull/7112

**Comment by Raibaz** [comment]
@TWiStErRob thanks for suggesting this, adopting the Kotlin binary compatibility validator makes a lot of sense.

**Comment by TWiStErRob** [comment]
@Raibaz do you think that https://github.com/mockk/mockk/issues/794#issuecomment-1086615157 would resolve this issue? (Before we go sidetrack and wait for a compatibility validator to get integrated. 🤓)

**Comment by Raibaz** [comment]
Yep, definitely, I added `languageVersion = 1.4` to build.gradle and the binary compatibility plugin.

**Comment by Raibaz** [comment]
Fixed by 9d19f15c841464a93e493f787d1ddacfac505886.
